lib/Job/Processor/HttpProcessor.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+namespace Resque\Job\Processor;
+
+use Firebase\JWT\JWT;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\BadResponseException;
+use GuzzleHttp\RequestOptions;
+use Resque\Protocol\HttpJobResponse;
+use Resque\Config\HttpProcessorConfig;
+use Resque\Job\JobParseException;
+use Resque\Job\RunningJob;
+use Resque\Log;
+use Resque\Protocol\Job;
+use Resque\Protocol\RunningLock;
+use Resque\Protocol\UniqueLockMissingException;
+use Resque\RedisError;
+use Resque\Resque;
+
+class HttpProcessor implements IProcessor {
+    public const PROCESSOR_NAME = 'Http';
+    
+    private HttpProcessorConfig $config;
+
+    public function __construct(HttpProcessorConfig $config) {
+        $this->config = $config;
+    }
+
+    public function process(RunningJob $runningJob): void {
+        try {
+            $this->runJob($runningJob);
+        } catch (\Throwable $t) {
+            try {
+                $runningJob->fail($t);
+                RunningLock::clearLock($runningJob->getJob()->getUniqueId());
+            } catch (\Throwable $r) {
+                Log::critical('Failed to properly handle job failure.', [
+                    Log::CTX_PROCESSOR => self::PROCESSOR_NAME,
+                    'exception' => $r,
+                    'payload' => $runningJob->getJob()->toArray()
+                ]);
+            }
+        }
+    }
+
+    private function runJob(RunningJob $runningJob): void {
+        try {
+            $job = $runningJob->getJob();
+
+            $client = new Client();
+
+            $url = str_replace('{sourceId}', $job->getSourceId(), $this->config->jobRunnerEndpointUrl);
+            $body = $job->toArray();
+
+            Log::debug('HTTP job request.', [
+                Log::CTX_PROCESSOR => self::PROCESSOR_NAME,
+                'jobId' => $job->getUniqueId(),
+                'request' => [
+                    'url' => $url,
+                    'body' => $body
+                ]
+            ]);
+
+            $headers = [
+                //TODO Authorization header could not be ready by the app.
+                //'Authorization' => "Bearer {$this->createBearerToken($job)}",
```

Comment:
is this TODO still relevant? we should reference the issue here or remove it

Reply:
Linked to the issue.